addon/components/file-upload/template.hbs

```diff
@@ -7,6 +7,7 @@
   disabled={{disabled}}
   onchange={{action "change" value="target.files"}}
   hidden
```

Comment:
Can these other attributes be removed in this template since they are now passed in as regular HTML attributes?
```
  accept={{accept}}
  capture={{capture}}	 
  multiple={{multiple}}	
  disabled={{disabled}}	 
```
Or perhaps they can just be given defaults in the template and then the `...attributes` will override them?

Reply:
This would be a breaking change as we won't support curly bracket invocation syntax anymore but I think that's fine as we prepare a new major anyways. @Alonski @gilest any thoughts on that one?

Reply:
The current implementation is classic component classes which support the angle bracket invocation syntax.

Important to remember that these are separate concepts in ember:
* **component type/class** (classic or glimmer)
* **component invocation** (curly or angle)

I think that classic components are a more flexible implementation for now since it supports both new and older ember apps.

If we use "tagless" glimmer components I think we may have difficulty supporting older versions of ember.

Is there any functionality or API that we are missing out on from not using `...attributes`?

Reply:
@gilest I fear you are missing inner-html vs outer-html in your concepts. Components using inner-html (`@ember/component` without `tagName: ''`) do not play well together with `...attributes` cause the HTML attributes passed on invocation would be set on the the element defined by component _and_ on the elements that have `...attributes`.

E.g. if you have a component `<FormElement>` like this:

```js
export default Component.extend({
  tagName: 'label',
});
```

```hbs
<input ...attributes>
```

And you invoke it with `<FormElement required />` the rendered HTML would be:

```html
<label required>
  <input required>
</label>
```

That's not what you want in most cases.

Reply:
You're right that's an important distinction! I think `tagName: ''` was kind of discouraged but I've used it a lot myself.